midterm-2023-11-07/code/5041-filip-zhikovski/scripts.ts

```diff
@@ -0,0 +1,202 @@
+
+const sortById =()=>
+{
+        const sortedMovies=movies.toSorted(idSorter);
+        displayMovies(sortedMovies)
+}
+const sortByTitle =()=>
+{
+    const sortedMovies=movies.toSorted(titleSorter);
+    displayMovies(sortedMovies)
+}
+
+const sortByYear =()=>
+{
+    const sortedMovies=movies.toSorted(yearSorter);
+    displayMovies(sortedMovies)
+}
+
+const sortByDirector =()=>
+{
+    const sortedMovies=movies.toSorted(directorSorter);
+    displayMovies(sortedMovies)
+}
+
+
+
+
+const applyNameFilter = () => {
+    const nameFilterInput = document.getElementById("name-filter") as HTMLInputElement;
+    const nameFilter = nameFilterInput.value.trim().toLowerCase();
+
+    let filteredMovies = movies;
+
+    if (nameFilter !== "") {
+        filteredMovies = movies.filter(movie => movie.title.toLowerCase().includes(nameFilter));
+    }
+
+    displayMovies(filteredMovies);
+};
+
+
+
+
+
+const loadData = async () => {
+    const url = 'https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-07/data/movies.json'
+    const response = await fetch(url);
+    if (!response.ok) {
+
+        throw new Error("Response Not OK")
+    }
+
+    const data = await response.json();
+    return data;
+}
+
+
+
+const displayMovies = (movies: Movie[]) => {
+
+    const container = document.getElementById('movie-container')!;
+    container.innerHTML = "";
+    for (const movie of movies) {
+        const movieRow = generateMovieRow(movie);
+        container.appendChild(movieRow);
+    }
+}
+  
+    const generateMovieRow = (movie: Movie) => {
+        const row = document.createElement('div');
+        row.classList.add("movie-table")
+    
+        const idCell = document.createElement('div');
+        idCell.classList.add('movie-data', 'movie-id');
+        idCell.innerHTML = movie.id.toString();
+        row.appendChild(idCell);
+    
+        const titleCell = document.createElement('div');
+        titleCell.classList.add('movie-data', 'movie-title');
+        titleCell.innerHTML = movie.title;
+        row.appendChild(titleCell);
+
+        const directorCell = document.createElement('div');
+        directorCell.classList.add('movie-data', 'movie-director');
+        directorCell.innerHTML = movie.director;
+        row.appendChild(directorCell);
+       
+        const yearCell = document.createElement('div');
+        yearCell.classList.add('movie-data', 'movie-year');
+        yearCell.innerHTML = movie.year.toString();
+        row.appendChild(yearCell);
+        
+        const genreCell = document.createElement('div');
+        genreCell.classList.add('movie-data', 'movie-genre');
+        genreCell.innerHTML = movie.genre.toString();
+        row.appendChild(genreCell);
+
+        const plotCell = document.createElement('div');
+        plotCell.classList.add('movie-data', 'movie-plot');
+
+        if (movie.plot.charAt(50) === ' ') { 
```

Comment:
You're not handling the case where the plot is less than 50 characters 😄 
Anyway, nice piece of code, but it should have been a separate function. +10